exporter/signalfxexporter/exporter.go

```diff
@@ -0,0 +1,213 @@
+			//  Or what others change from default values?
+			Timeout: config.Timeout,
+		},
+		logger: logger,
+		zippers: sync.Pool{New: func() interface{} {
+			return gzip.NewWriter(nil)
+		}},
+	}
+
+	exp, err := exporterhelper.NewMetricsExporter(
+		&config.ExporterSettings,
+		s.pushMetricsData,
+		exporterhelper.WithTracing(true),
+		exporterhelper.WithMetrics(true))
+
+	return exp, err
+}
+
+// httpSender sends the data to the SignalFx backend.
+type httpSender struct {
+	url     string
+	headers map[string]string
+	client  *http.Client
+	logger  *zap.Logger
+	zippers sync.Pool
+}
+
+func (s *httpSender) pushMetricsData(
+	ctx context.Context,
+	md consumerdata.MetricsData,
+) (droppedTimeSeries int, err error) {
+
+	sfxDataPoints, numDroppedTimeseries, err := metricDataToSingalFxV2(s.logger, md)
+	if err != nil {
+		return exporterhelper.NumTimeSeries(md), consumererror.Permanent(err)
+	}
+
+	body, compressed, err := s.encodeBody(sfxDataPoints)
+	if err != nil {
+		return exporterhelper.NumTimeSeries(md), consumererror.Permanent(err)
+	}
+
+	req, err := http.NewRequest("POST", s.url, body)
+	if err != nil {
+		return exporterhelper.NumTimeSeries(md), consumererror.Permanent(err)
+	}
+
+	if s.headers != nil {
+		for k, v := range s.headers {
+			req.Header.Set(k, v)
+		}
+	}
+
+	if compressed {
+		req.Header.Set("Content-Encoding", "gzip")
+	}
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return exporterhelper.NumTimeSeries(md), err
+	}
+
+	io.Copy(ioutil.Discard, resp.Body)
+	resp.Body.Close()
+
+	// SignalFx accepts all 2XX codes.
+	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
+		err = fmt.Errorf(
+			"HTTP %d %q",
+			resp.StatusCode,
+			http.StatusText(resp.StatusCode))
+		return exporterhelper.NumTimeSeries(md), err
+	}
+
+	return numDroppedTimeseries, nil
+}
+
+func buildHeaders(config *Config) (map[string]string, error) {
+	if config.AuthToken == "" {
+		err := fmt.Errorf(
+			"%q config requires a non-empty \"auth_token\"",
+			config.Name())
+		return nil, err
+	}
+
+	headers := map[string]string{
+		"Connection":   "keep-alive",
+		"Content-Type": "application/x-protobuf",
+		"User-Agent":   "OpenTelemetry-Collector SignalFx Exporter/v0.0.1",
+		"X-Sf-Token":   config.AuthToken,
+	}
+
+	// Add any custom headers from the config. They will override the pre-defined
+	// ones above in case of conflict, but, not the content encoding one since
+	// the latter one is defined according to the payload.
+	for k, v := range config.Headers {
+		headers[k] = v
+	}
+
+	return headers, nil
+}
+
+func (s *httpSender) encodeBody(dps []*sfxpb.DataPoint) (bodyReader io.Reader, compressed bool, err error) {
+	msg := &sfxpb.DataPointUploadMessage{
+		Datapoints: dps,
+	}
+	body, err := proto.Marshal(msg)
+	if err != nil {
+		return nil, false, err
+	}
+	return s.getReader(body)
+}
+
+// avoid attempting to compress things that fit into a single ethernet frame
```

Comment:
I took this from https://github.com/signalfx/golib/blob/f517aa507cde4806d40b0cd5eb1ad1f2555eda91/sfxclient/httpsink.go#L315

@keitwb do you have any info on the question above?

Reply:
I think the intent with it was more just to avoid compressing very small payloads since for very small payloads the overhead of compression can be worse than the benefits of it.  The threshold for it being inefficient might be much lower than 1500 bytes though, I'm not sure.  You can probably just compress everything since CPU usage won't really be an issue when there are small payloads anyway.

Reply:
I am fine to keep as is since this is ported from a production tested code.